spmd/tensor/dispatch.py

```diff
@@ -22,7 +22,7 @@
 """
 If set to true, __DEBUG_STRICT will fail when an op doesn't have a sharding rule registered.
 """
-_DEBUG_STRICT = False
+_DEBUG_STRICT = True
```

Comment:
Should we remove this?

Reply:
yeah I think we can remove it, but i guess we will have a similar flag later to enable the fallback of redistribute to replicate, let me change the name of this.

Reply:
gonna merge this PR first, we can use this https://github.com/pytorch/tau/pull/413 to increase test coverage